### PR TITLE
Add Regexp for text type attribute value validation

### LINKED ISSUE #1493: Add Regexp for attribute value validation
**What is the expected behavior?**
When adding a new content of type 'number', in content type builder.
min/max 'value' is present as an option but 'length' isn't. I'm aware they could conflict against one another, perhaps allowing one or the other?

Bug or feature? I don't know. I tried to solve it my self [but](https://github.com/strapi/strapi/issues/415#issuecomment-401563765).

### PR DESCRIPTION
**Pull request by yacir**
FIX #1493

WIP: If it LGTY, translations should be added before merge.

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [ ] 💅 Enhancement
- [x] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [x] Plugin

#### Manual testing done on the following databases:

- [ ] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

#### Result:
Capture d’écran 2020-01-07 à 12 57 59

### CODE REVIEW

File: packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/forms.js [hunk 1]
```diff
@@ -235,6 +235,12 @@ const forms = {
               ),
             enumName: yup.string().nullable(),
           });
+        case 'text':
+          return yup.object().shape({
+            ...commonShape,
+            ...fieldsThatSupportMaxAndMinLengthShape,
+            regex: yup.string().nullable(),
```
Comment (alexandrebodin) [inline packages/strapi-plugin-content-type-builder/admin/src/containers/FormModal/utils/forms.js]: You need to add a test to check the regex value is valid. check that you can create a new Regexp without it throwing should be enough

File: packages/strapi-plugin-content-type-builder/controllers/validation/types.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ const getTypeShape = (attribute, { modelType } = {}) => {
         unique: validators.unique,
         minLength: validators.minLength,
         maxLength: validators.maxLength,
+        regex: yup.string(),
```
Comment (alexandrebodin) [inline packages/strapi-plugin-content-type-builder/controllers/validation/types.js]: Same here :)

Comment (yacir) [reply packages/strapi-plugin-content-type-builder/controllers/validation/types.js]: something like 

```js 
const isValidRegExpPattern = {
  name: 'isValidRegExpPattern',
  message: '${path} must be a valid RexExp pattern string',
  test: val => val === '' ||  new RegExp(val),
};
```

**Comment by alexandrebodin** [comment]
We will ba adding some validation layer in the APIs before merging this ;) Great work anyway !

**Comment by yacir** [comment]
> We will ba adding some validation layer in the APIs before merging this ;) Great work anyway !

Let me know if I can add those validations in  this MR :)

**Comment by alexandrebodin** [comment]
@yacir We are introducing the validation layer really soon :) This feature would be really cool to add to it ! I will let you know as soon as it is merged and you can start working on it :D

**Comment by alexandrebodin** [comment]
Hi @yacir We release the single-type feature in beta.19. With it we released an entityValidator where you could add your regex validation logic so the backend validates it too :) If you feel like updating your PR ;)

**Comment by alexandrebodin** [comment]
Hi,

I'm going to close this PR has it is getting really outdated. 

If you ever get time to update it it would be really cool :D

**Pull request closed without merging**